src/plugins/dashboard/public/dashboard_container/external_api/dashboard_renderer.tsx

```diff
@@ -59,6 +59,17 @@ export const DashboardRenderer = forwardRef<AwaitingDashboardAPI, DashboardRende
     const [dashboardContainer, setDashboardContainer] = useState<DashboardContainer>();
     const [fatalError, setFatalError] = useState<ErrorEmbeddable | undefined>();
     const [dashboardMissing, setDashboardMissing] = useState(false);
+    const [useMargins, setUseMargins] = useState(dashboardContainer?.getInput().useMargins);
+
+    useEffect(() => {
+      if (dashboardContainer?.getInput$) {
+        const s = dashboardContainer?.getInput$().subscribe((value) => {
+          setUseMargins(value.useMargins);
+        });
+
+        return () => s.unsubscribe();
+      }
+    }, [dashboardContainer]);
```

Comment:
Curious - could we not move this logic into `dashboard_viewport` so that we could do this instead?

```typescript
const useMargins = dashboard.select((state) => state.explicitInput.useMargins);
``` 

Here's a very quick mockup of what that looks like (with very ugly colours):


https://github.com/elastic/kibana/assets/8698078/4e51fb66-a515-4830-94ba-b84652df6611


And this is the styling I used:

```typescript
      <div
        ...
        style={
          useMargins
            ? { height: '100%', background: 'red' }
            : { height: '100%', background: 'green' }
        }
      >
        {viewportWidth !== 0 && <DashboardGrid viewportWidth={viewportWidth} />}
      </div>
```

(obviously, the above would need to be cleaned up with classes rather than inline styling)

Reply:
@Heenawter Thanks for the pointer, I some how missed the viewport wrapper this is a much better approach than actually opting to subscribe to the entire dashboard input!!